tools/codegen/api/translate.py

```diff
@@ -194,6 +195,28 @@ def direct_solve(goal: NamedCType) -> str:
             options = direct_solve(options_ctype)
             return f'{options}.pinned_memory_opt()'
 
+        # Note [translation from C++ reference to value types]
+        # The below cases are all for when we have an argument with a reference type,
+        # and a corresponding goal with a value type.
+        # These are needed when we populate the inputs to a lambda capture and we need
+        # to guarantee the lifetime of each captured argument.
+        elif goal.type == VectorCType(BaseCType(intT)):
```

Comment:
Not sure why you would specialize the rule here to work only on intT

Reply:
This is a potential performance hole, because vec() requires an O(n) allocation of an entirely new vector to serve as the owning reference. It might be safer to only enable this conversion if someone passes a kwarg saying `allow_expensive_conversions`

Reply:
agreed, adding a kwarg to translate() sounds reasonable